source/README.adoc

```diff
@@ -0,0 +1,171 @@
+
+:toc:
+:toc-placement!:
+= RabbitMQ Source
+
+The RabbitMQ source translates messages on a RabbitMQ exchange to CloudEvents,
+which can then be used with Knative Eventing over HTTP. The source can bind to
+an existing RabbitMQ exchange, or create a new exchange if required.
+
+toc::[]
+
+
+== Published events
+
+All messages received by the source are published with the following schema:
+
+.Event attributes
+|===
+| Attribute | Value | Notes
+
+| `type` | `dev.knative.rabbitmq.event` |
+| `source` | `/apis/v1/namespace/*$NS*/rabbitmqsources/*$NAME*#*$TOPIC*`
+   | `NS`, `NAME` and `TOPIC` are derived from the source configuration
+| `id` | A unique ID | This uses the `MessageId` if available, and a UUID otherwise
+| `subject` | The ID of the message | Empty string if no message ID is present
+| `datacontenttype` | `application/json` | Currently static
+| `key` | The ID of the message | Empty string if no message ID is present
+|===
+
+The payload of the event is set to the data content of the message.
```

Comment:
```suggestion
[source,json]
----
{
  "specversion": "1.0",
  "type": "dev.knative.rabbitmq.event",
  "source": "/apis/v1/namespaces/<namespace>/rabbitmqsources/<name>#<topic>", <1>
  "id": "<unique_ID>", <2>
  "time": "<timestamp>",
  "subject": "", <3>
  "datacontenttype": "application/json",
  "key": "", <4>
  "data": "Hello rabbitmq!" <5>
}
----
<1> `<namespace>`, `<name>` and `<topic>` are derived from the source configuration.
<2> Uses the `MessageId` if available, and a UUID otherwise.
<3> The ID of the message, or an empty string if no message ID is present.
<4> The ID of the message, or an empty string if no message ID is present.
<5> The payload of the event is set to the data content of the message.
```

Reply:
Rather than the table, I'd suggest showing the actual schema as an example like this, where there are callouts describing any variables

Reply:
So no table at all? I have the actual schema in the example, though with values filled in.